src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,454 @@
+        require(may[msg.sender] == 1 || may[address(0)] == 1, "RwaMultiSwapOutputConduit/not-mate");
+        _;
+    }
+
+    /**
+     * @notice Defines addresses and gives `msg.sender` admin access.
+     * @param _dai DAI contract address.
+     */
+    constructor(address _dai) public {
+        require(_dai != address(0), "RwaMultiSwapOutputConduit/wrong-dai-address");
+
+        dai = DaiAbstract(_dai);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` operator access to this contract.
+     * @param usr The user address.
+     */
+    function hope(address usr) external auth {
+        can[usr] = 1;
+        emit Hope(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` operator access from this contract.
+     * @param usr The user address.
+     */
+    function nope(address usr) external auth {
+        can[usr] = 0;
+        emit Nope(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /**
+     * @notice Whitelist `who` address for `pick`.
+     * @param who The user address.
+     */
+    function kiss(address who) external auth {
+        bud[who] = 1;
+        emit Kiss(who);
+    }
+
+    /**
+     * @notice Remove `who` address from `pick` whitelist.
+     * @param who The user address.
+     */
+    function diss(address who) external auth {
+        if (to == who) to = address(0);
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    /**
+     * @notice Whitelist `psm` address for `hook`.
+     * @param _psm The PSM address.
+     */
+    function clap(address _psm) external auth {
+        require(PsmAbstract(_psm).dai() == address(dai), "RwaMultiSwapOutputConduit/wrong-dai-for-psm");
+
+        // Check if GEM decimals is not greater then DAI decimals. We assume that DAI will alway have 18 decimals
+        // Does decimals should be less then 18, according to AuthGemJoin5 check (decimals < 18)
+        _sub(18, GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem()).decimals());
+
```

Comment:
I believe we don't need this check here to be honest.

This will be checked down the line regardless, so there is no harm in having a `clap`ped PSM with a gem whose `decimals` are higher than 18.
```suggestion
        // Check if GEM decimals is not greater then DAI decimals. We assume that DAI will alway have 18 decimals
        // Does decimals should be less then 18, according to AuthGemJoin5 check (decimals < 18)
        _sub(18, GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem()).decimals());

```

Reply:
I agree, just keep that here as we have it in SwapConduit

Reply:
I see what you mean. In this case, I'd suggest we change the comment for clarity:

```suggestion
        // Check if GEM decimals is not greater then DAI decimals. 
        // We assume that DAI will alway have 18 decimals.
        // The current PSMs gem join `AuthGemJoin5` already checks if `decimals <= 18`
        // However for the sake of correction, we perform the same check here.
        // This is also consistent with the behavior of `RwaSwapOutputConduit`
        _sub(18, GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem()).decimals());

```